### PR TITLE
Fix: #19, Fix: Ci, Fix: test in windows

### PR DESCRIPTION
**Pull request by yihong0618**
What does this implement/fix? Explain your changes.
---------------------------------------------------
Fix: #19, Fix: Ci, Fix: test db file in windows (but do a little change please check)

Does this close any currently open issues?
---------------------------------------------------
Yes

### CODE REVIEW

**Comment by yihong0618** [comment]
rebase  in one commit

File: kb/commands/add.py [hunk 1]
```diff
@@ -70,9 +70,13 @@ def add(args: Dict[str, str], config: Dict[str, str]):
...
             # otherwise open up the editor and create some content
-            shell_cmd = shlex.split(config["EDITOR"]) + [Path(category_path, title)]
-            call(shell_cmd)
-
+            shell_cmd = shlex.split(config["EDITOR"]) + [str(Path(category_path, title))]
+            if sys.platform.startswith('win'):
+                # when windows need shell True
+                call(shell_cmd, shell=True)
+            else:
+                call(shell_cmd)
```
Comment (cclauss) [inline kb/commands/add.py:78]:
```suggestion
            # Windows need shell=True
            call(shell_cmd, shell=sys.platform.startswith('win'))
```

**Comment by gnebbia** [comment]
But is that shell=True needed?

**Comment by yihong0618** [comment]
> But is that shell=True needed?

Seems you are right.
I tested and it seems not need, I will take another commit.

File: kb/commands/add.py (hunk 1 above)
Comment (yihong0618) [reply kb/commands/add.py:78]: Nice catch, but it seems that I can not reproduce it

**Comment by gnebbia** [comment]
Ok I will wait for the commit and then merge!

**Comment by cclauss** [comment]
It is more secure not to use `shell=True`. https://docs.python.org/3/library/subprocess.html#security-considerations

**Comment by gnebbia** [comment]
@cclauss indeed this is what I was saying, I prefer not to use shell=True, since I saw that you were using it, I waited before merging all of your changes...

Since it works also without shell=True, I would definitely say to remove it.

**Comment by yihong0618** [comment]
> secure



> It is more secure not to use `shell=True`. https://docs.python.org/3/library/subprocess.html#security-considerations

Learned that, Thank you guys.

**Comment by yihong0618** [comment]
@gnebbia  @cclauss  Done

**Comment by gnebbia** [comment]
Thanks a lot Sir!

**Comment by yihong0618** [comment]
@gnebbia I reproduce it.

If we add default editor like `code` `vim` for windows, we need to add shell=True.
But I think you are right secure  is more important.

**Comment by gnebbia** [comment]
@yihong0618 ok so what do you suggest? if we keep "code" then it does not work without shell=True ?
Don't you have a suggestion for common windows editors?

**Comment by yihong0618** [comment]
> @yihong0618 ok so what do you suggest? if we keep "code" then it does not work without shell=True ?
> Don't you have a suggestion for common windows editors?

I think you can keep the origin code, `code` is useless for now.

Can we print a message that tell win user to set  **EDITOR**?

**Comment by gnebbia** [comment]
@yihong0618 I was thinking about putting some documentation in the README.md that says how to set the %EDITOR% variable by command line.

What do you think?
Ok so I will remove the default editor stuff!

**Comment by yihong0618** [comment]
> @yihong0618 I was thinking about putting some documentation in the README.md that says how to set the %EDITOR% variable by command line.
> 
> What do you think?
> Ok so I will remove the default editor stuff!

I think that will be good~

**Comment by yihong0618** [comment]
Do we need close the conn using `with statment` or `conn.close()` in add, or edit file like in `test_db`

**Comment by gnebbia** [comment]
I think the with statement is better!
But tests are still a mess, we are planning to complete them all in these days.

**Comment by yihong0618** [comment]
> I think the with statement is better!
> But tests are still a mess, we are planning to complete them all in these days.

Yes, I also found something bug today. - -.
And I think rm all the data in `tests/data` seems better.

**Comment by gnebbia** [comment]
Please update frequently... I just today updated the tests!
Let me know if you still have the bug, and how to reproduce it!
Thanks a lot!

**Pull request merged**